python/cudf/cudf/core/column/lists.py

```diff
@@ -725,3 +725,57 @@ def astype(self, dtype):
                 lambda col, dtype: col.astype(dtype), dtype
             )
         )
+
+    def fillna(self, value, method=None):
+        """
+        Replaces None and nan at all leaf levels with the specified value
+        or replacement method.
+
+        Parameters
+        ----------
+        value: the value to replace None and nan with.
+        method (optional): the method to use to replace None values.
+
+        Returns
+        -------
+        A new list without None or nan values.
+
+        Notes
+        -----
+        value and method cannot be specified simultaneously.
+
+        Using this function on Lists with varying nesting levels containing
+        None or nan will result in an error. The list type specification is
+        more flexible for lists containing None than those without.
+
+        Examples
+        --------
+        >>> s = cudf.Series([[1, 2], [None, 4]])
+        >>> s2 = s.list.fillna(3)
+        >>> s2
+        0    [1, 2]
+        1    [3, 4]
+        dtype: list
+        """
+        if value is not None and method is not None:
+            raise ValueError("Cannot specify both 'value' and 'method'.")
+
+        if method:
+            if method not in {"ffill", "bfill", "pad", "backfill"}:
+                raise NotImplementedError(
+                    f"Fill method {method} is not supported"
+                )
+            if method == "pad":
+                method = "ffill"
+            elif method == "backfill":
+                method = "bfill"
+        result = self._column.copy(deep=True)
+        current = result
+        while len(current._base_children) != 0:
+            new_children = [child for child in current._base_children[:-1]]
+            new_children.append(
+                current._base_children[-1].fillna(value, method)
```

Comment:
If `base_children[-1]` is a `ListColumn` how does this work? AFAICT, `fillna` is implemented on all column types but raises a libcudf error if not supported (which it is not on `ListColumn`s).

Reply:
It doesn't raise an error, it does nothing when `base_children[-1]` is a `ListColumn` IIRC.

```
c = cudf.Series([[1, None], [2, None], [3, None]])
c._column.fillna(1)
```
No error, no change.